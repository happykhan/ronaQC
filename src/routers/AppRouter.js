import React, { useReducer } from "react";
import { Route, Routes, BrowserRouter } from "react-router-dom";
import ControlPage from "../components/ControlPage";
import SampleReportPage from "../components/SampleReportPage";
import ImportPage from "../components/ImportPage";
import HelpPage from "../components/HelpPage";
import NotFoundPage from "../components/NotFoundPage";
import Layout from "../components/Layout";
import NegativeControlContext from "../context/NegativeControlContext";
import NegativeControlReducer from "../reducers/NegativeControlReducer";
import SampleReducer from "../reducers/SampleReducer";
import SampleContext from "../context/SampleContext";

const AppRouter = () => {
  const [negativeControl, dispatch] = useReducer(NegativeControlReducer, []);
  const [samples, sampleDispatch] = useReducer(SampleReducer, []);
  return (
    <BrowserRouter>
      <Layout>
        <NegativeControlContext.Provider value={{ negativeControl, dispatch }}>
          <SampleContext.Provider value={{ samples, sampleDispatch }}>
            <Routes>
              <Route path="/" element={<ImportPage />} />
              <Route path="control" element={<ControlPage />} />
              <Route path="report" element={<SampleReportPage />} />
              <Route path="help" element={<HelpPage />} />
              <Route path="*" element={<NotFoundPage />} />
            </Routes>
          </SampleContext.Provider>
        </NegativeControlContext.Provider>
      </Layout>
    </BrowserRouter>
  );
};

export default AppRouter;
