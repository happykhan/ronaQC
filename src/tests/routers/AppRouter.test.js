/**
 * @jest-environment jsdom
 */
import React from "react";
import AppRouter from "../../routers/AppRouter";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

test("should render Full site (import page) correctly", () => {
  render(<AppRouter />);
  expect(screen.getByTestId(`Import-page`)).toMatchSnapshot();
});
