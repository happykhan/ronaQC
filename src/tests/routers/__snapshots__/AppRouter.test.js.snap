// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Full site (import page) correctly 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  data-testid="Import-page"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1h1zemz-MuiTypography-root"
      >
        Import
      </h3>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-qwfiov-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1upwz12-MuiTypography-root"
          >
            RonaQC accepts mapped SARS-CoV-2 reads (BAM format), generated from the SARS-CoV-2 bioinformatic pipelines like ARTIC, and any control samples from the respective sequencing run (negative/positive) as input. It will then assess the levels of cross contamination and primer contamination in the samples, and determine if the samples are reliable for detecting SARS-CoV-2, phylogenetic analysis, and/or submission to public databases.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
        >
          <form>
            <input
              type="file"
            />
            <span
              class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="submit"
            >
              Upload Negative control
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </form>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
        >
          <form>
            <input
              multiple=""
              type="file"
            />
            <span
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              role="button"
              tabindex="0"
              type="submit"
            >
              Upload Sample data
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </form>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Demo
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Demo (repeat analysis)
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
